src/stream_inserter.h

```diff
@@ -12,46 +12,87 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#include <greptime/v1/database.grpc.pb.h>
-#include <grpcpp/channel.h>
+#include <queue>
+#include <mutex>
+#include <atomic>
 #include <memory>
-#include "grpcpp/client_context.h"
+#include <thread>
+#include <cassert>
+#include <cstddef>
+#include <iostream>
+#include <condition_variable>
+
+#include <grpcpp/channel.h>
+#include <grpcpp/client_context.h>
+#include <grpcpp/support/sync_stream.h>
+
+#include <greptime/v1/database.pb.h>
+#include <greptime/v1/database.grpc.pb.h>
+
 namespace greptime {
 
-using String = std::string;
 using greptime::v1::GreptimeDatabase;
 using greptime::v1::GreptimeRequest;
 using greptime::v1::GreptimeResponse;
 using greptime::v1::InsertRequest;
 using greptime::v1::InsertRequests;
+using greptime::v1::RequestHeader;
 using grpc::ClientContext;
 using grpc::Status;
 using grpc::Channel;
+using grpc::ClientWriter;
 
 class StreamInserter {
 public:
-    StreamInserter(String dbname_, std::shared_ptr<Channel> channel_, std::shared_ptr<GreptimeDatabase::Stub> stub_) :
-    dbname(dbname_), channel(channel_), stub(stub_) {
+    StreamInserter(std::string dbname_, std::shared_ptr<Channel> channel_, std::shared_ptr<GreptimeDatabase::Stub> stub_) :
+            dbname(dbname_),
+            channel(channel_),
+            stub(stub_) {
         context = std::make_shared<ClientContext>();
         context->set_wait_for_ready(true);
         writer = std::move(stub_->HandleRequests(context.get(), &response));
+        scheduler_thread_is_running = true;
+        scheduler_thread = new std::thread(&StreamInserter::RunHandleRequest, this);
     }
 
-    bool Write(InsertRequests &insert_requests);
+    ~StreamInserter() {
+        delete scheduler_thread;
+    }
+
+    void Write(InsertRequest insert_request);
+
+    void WriteBatch(std::vector<InsertRequest> insert_request_vec);
 
-    bool WriteDone() { return writer->WritesDone();}
+    bool Send(const GreptimeRequest &greptime_request);
+
+    void RunHandleRequest();
+
+    bool WriteDone() {
```

Comment:
What's the difference between `WriteDone` and `Finish`? Can we explicitly doc these two methods?